src/app/common/services/global-app-intercepter.service.ts

```diff
@@ -0,0 +1,50 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { AuthService } from 'src/app/services/auth.service';
+
+@Injectable()
+export class GlobalAppIntercepter implements HttpInterceptor {
+    constructor(private authService: AuthService) {}
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        let authToken;
+        const commonHeaders = {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PATCH, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Origin, Content-Type, X-Auth-Token'
+        };
+        if (req.url.indexOf('login') !== -1) {
```

Comment:
Can we use switch case instead of if else?

Reply:
we can. but I think it doesn't make much difference here as we only have 2 conditions and we are not comparing with any fix value